app/build.gradle

```diff
@@ -29,11 +34,11 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_17
```

Comment:
Is there a reason for JDK level increase?

Reply:
latest compose version needed a more recent compose compiler which needed a more recent Kotlin version which asked for a more recent jdk  so I just kept choosing the latest versions to get over the "not compatible versions" warnings and errors

short version: a series of incompatible versions led me to choose the latest jdk